File: TaskyJ.Interface.Angular/ClientApp/src/app/models/DBCategoryJ.ts
import { BaseEntity } from './BaseEntity';

export class DBCategoryJ implements BaseEntity {
  id: number;
  name: string = null;
  icon: string = null;
  
  public Equals(source: DBCategoryJ): boolean {
    return this.id == source.id &&
      this.name == source.name;
  }

  public CopyFrom(source: DBCategoryJ): void {
    this.id = source.id;
    this.name = source.name;
    this.icon = source.icon;
  }
}

File: TaskyJ.Globals.Data/DataObjects/DBCategoryJ.cs
using System.Runtime.Serialization;
using TaskyJ.Globals.Data.Helpers;

namespace TaskyJ.Globals.Data.DataObjects
{
    public class DBCategoryJ : BaseEntity
    {
        [DataMember]
        public string Name { get; set; }
        [DataMember]
        public string IconBase64
        {
            get;
            set;
        }

        public TaskyJImage Icon
        {
            get
            {
                return ImageHelper.Base64StringToImage(IconBase64);
            }
            set
            {
                IconBase64 = ImageHelper.ImageToBase64(value);
            }
        }

        public override string ToString()
        {
            return Name;
        }

        public override bool Equals(BaseEntity source)
        {
            if (source is DBCategoryJ sourcej)
            {
                return Name == sourcej.Name && base.Equals(source);
            }
            else
            {
                return base.Equals(source);
            }
        }
    }
}
